baybe/constraints/conditions.py

```diff
@@ -129,6 +151,13 @@ def evaluate(self, data: pd.Series) -> pd.Series:  # noqa: D102
 
         return data.apply(func)
 
+    def convert_to_polars(self) -> Callable:
+        """Convert the condition to a callable for Polars."""
+        func = rpartial(_threshold_operators[self.operator], self.threshold)
+        if self.operator in _valid_tolerance_operators:
+            func = rpartial(func, atol=self.tolerance)
+        return func
```

Comment:
I have my problems with this part:
* The code is 100% copy-pasted from the lines above, which should be avoided. The original code should thus use this new function as well, but for that a rename needs to happen (see next point)
* I cannot see how the function body is any related to "polars" – it's just operator preparation stuff which also works in non-polar contexts and in fact is used in numpy context, since you copied the code from above. Hence, the function name makes not too much sense to me
* The output type should be more specific. What are the inner types of that `Callable`?

Reply:
- [ ] Refactor `evaluate` and `convert_to_polars`.

> I cannot see how the function body is any related to "polars" – it's just operator preparation stuff which also works in non-polar contexts and in fact is used in numpy context, since you copied the code from above.

Another version of this function is also implemnted in `SubSelectionCondition` - it uses Polars, but not pushed here because they're related to `DiscreteExclusionConstraint`. 

> The output type should be more specific. What are the inner types of that Callable?
Output type changed to `pl.Expr`